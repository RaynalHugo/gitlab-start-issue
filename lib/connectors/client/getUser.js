"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _fp = require("lodash/fp");

function getUser() {
  return _getUser.apply(this, arguments);
}

function _getUser() {
  _getUser = (0, _asyncToGenerator2.default)(
  /*#__PURE__*/
  _regenerator.default.mark(function _callee() {
    var client, store, user;
    return _regenerator.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            client = this.client;
            store = this.store;

            if (!(0, _fp.isNil)(client)) {
              _context.next = 6;
              break;
            }

            throw new Error("no client in 'getUser'");

          case 6:
            if (!(0, _fp.isNil)(store)) {
              _context.next = 10;
              break;
            }

            throw new Error("no store in 'getUser'");

          case 10:
            _context.t0 = _fp.get;
            _context.next = 13;
            return client.get("/user");

          case 13:
            _context.t1 = _context.sent;
            user = (0, _context.t0)("data", _context.t1);
            _context.next = 17;
            return store.setItem("user", user);

          case 17:
            return _context.abrupt("return", user);

          case 18:
          case "end":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));
  return _getUser.apply(this, arguments);
}

var _default = getUser;
exports.default = _default;