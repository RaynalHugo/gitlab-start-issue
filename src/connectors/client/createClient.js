import axios from "axios";
import { isNil } from "lodash/fp";

export function createClient({ userPrivateToken, gitlabUrl }) {
	if (!isNil(userPrivateToken)) {
		return axios.create({
			baseURL: gitlabUrl,
			timeout: 10000,
			headers: { "Private-Token": userPrivateToken }
		});
	} else {
		throw new Error(
			"userPrivateToken is not definied in client `createClient`"
		);
	}
}

export const client = createClient({
	gitlabUrl: "https://gitlab.com/api/v4",
	userPrivateToken: "ddx1wWi8xo6ryc4XbBqj"
});
