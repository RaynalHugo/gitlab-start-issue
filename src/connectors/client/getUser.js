import { get, isNil } from "lodash/fp";

async function getUser() {
	const client = this.client;
	const store = this.store;
	if (isNil(client)) {
		throw new Error("no client in 'getUser'");
	} else if (isNil(store)) {
		throw new Error("no store in 'getUser'");
	} else {
		const user = get("data", await client.get(`/user`));
		console.log(user);
		await store.setItem("user", user);
		return user;
	}
}

export default getUser;

async function getUser2(client) {
	return get("data", await client.get(`/user`));
}

export { getUser2 };
