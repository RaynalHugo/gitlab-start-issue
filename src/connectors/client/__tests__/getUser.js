import { getUser2 } from "../getUser";
import { client } from "../createClient";

test("getUser2", async () => {
	expect(await getUser2(client)).toMatchSnapshot();
});
