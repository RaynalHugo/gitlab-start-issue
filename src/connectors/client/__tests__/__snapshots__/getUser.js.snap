// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUser2 1`] = `
Object {
  "avatar_url": "https://assets.gitlab-static.net/uploads/-/system/user/avatar/3058287/avatar.png",
  "bio": "",
  "can_create_group": true,
  "can_create_project": true,
  "color_scheme_id": 1,
  "confirmed_at": "2018-10-29T16:20:28.744Z",
  "created_at": "2018-10-29T16:20:20.833Z",
  "current_sign_in_at": "2019-01-30T22:13:16.862Z",
  "email": "hugo.raynal@sterblue.com",
  "external": false,
  "id": 3058287,
  "identities": Array [],
  "last_activity_on": "2019-01-30",
  "last_sign_in_at": "2019-01-29T10:57:09.542Z",
  "linkedin": "https://www.linkedin.com/in/hugoraynal/",
  "location": "Nantes",
  "name": "Hugo Raynal",
  "organization": "Sterblue",
  "private_profile": false,
  "projects_limit": 100000,
  "public_email": "",
  "shared_runners_minutes_limit": null,
  "skype": "",
  "state": "active",
  "theme_id": 1,
  "twitter": "",
  "two_factor_enabled": false,
  "username": "hugo.raynal",
  "web_url": "https://gitlab.com/hugo.raynal",
  "website_url": "",
}
`;
